spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/sampler/TraceIdToBigIntegerConverter.java

```diff
@@ -0,0 +1,24 @@
+package org.springframework.cloud.sleuth.sampler;
+
+import java.math.BigInteger;
+import java.util.UUID;
+
+class TraceIdToBigIntegerConverter {
+
+	/**
+	 * Takes two sets of bits (most and least significant ones), puts them together
+	 * and represents as a BigInteger
+	 *
+	 * @param traceIdToConvert
+	 * @return - BigInteger representation of the TraceId
+	 */
+	static BigInteger traceIdToBigInt(String traceIdToConvert) {
```

Comment:
I've an idea about how to eliminate this btw.., instead of using BigInteger, just compare 2 longs (or 4 ints)